propagator/opentelemetry-propagator-ot-trace/tests/test_ot_trace_propagator.py

```diff
@@ -0,0 +1,360 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from unittest import TestCase
+
+from opentelemetry.baggage import get_all, set_baggage
+from opentelemetry.propagator.ot_trace import (
+    OT_BAGGAGE_PREFIX,
+    OT_SAMPLED_HEADER,
+    OT_SPAN_ID_HEADER,
+    OT_TRACE_ID_HEADER,
+    OTTracePropagator,
+)
+from opentelemetry.sdk.trace import _Span
+from opentelemetry.trace import (
+    INVALID_SPAN_CONTEXT,
+    INVALID_SPAN_ID,
+    INVALID_TRACE_ID,
+    SpanContext,
+    TraceFlags,
+    set_span_in_context,
+)
+from opentelemetry.trace.propagation import get_current_span
+from opentelemetry.trace.propagation.textmap import DictGetter
+
+carrier_getter = DictGetter()
+
+
+class TestOTTracePropagator(TestCase):
+
+    ot_trace_propagator = OTTracePropagator()
+
+    def carrier_inject(self, trace_id, span_id, is_remote, trace_flags):
+
+        carrier = {}
+
+        self.ot_trace_propagator.inject(
+            dict.__setitem__,
+            carrier,
+            set_span_in_context(
+                _Span(
+                    "child",
+                    context=SpanContext(
+                        trace_id=trace_id,
+                        span_id=span_id,
+                        is_remote=is_remote,
+                        trace_flags=trace_flags,
+                    ),
+                )
+            ),
+        )
+
+        return carrier
+
```

Comment:
Could add an `inject()` test case for trace ID with fewer than 32 hex chars, and span ID with fewer than 16?

Reply:
Fixed